packages/teuchos/comm/src/Teuchos_StackedTimer.cpp

```diff
@@ -0,0 +1,204 @@
+// @HEADER BEGIN
+// @HEADER END
+
+#include "Teuchos_StackedTimer.hpp"
+#include <limits>
+
+
+namespace Teuchos {
+
+Teuchos::RCP<StackedTimer> StackedTimer::timer;
+  
+StackedTimer::LevelTimer::LevelTimer() :
+    level_(std::numeric_limits<unsigned>::max()),name_("INVALID"),parent_(NULL)
+{}
+
+void error_out(const std::string& msg, const bool)
+{
+  TEUCHOS_TEST_FOR_EXCEPTION(true,std::runtime_error,msg);
+}
+  
+
+void
+StackedTimer::LevelTimer::report(std::ostream &os) {
+  for (unsigned i=0; i<level_; ++i)
+    os << "   ";
+  os << name_<<":"<<accumulatedTime()<< " [" << count_started_<<"] ("<< count_updates_ <<")"<<std::endl;
+  double t_total = 0;
+  for (size_t i=0; i<sub_timers_.size(); ++i) {
+    t_total += sub_timers_[i].accumulatedTime();
+    sub_timers_[i].report(os);
+  }
+  if ( sub_timers_.size() == 0 )
+    return;
+  for (unsigned i=0; i<=level_; ++i)
+    os << "   ";
+  os << "Remainder: " << accumulatedTime() - t_total<<std::endl;
+
+}
+
+BaseTimer::TimeInfo
+StackedTimer::LevelTimer::findTimer(std::string &name) {
```

Comment:
Should pass in the string by const reference or by value.  Passing in by nonconst reference will make reasonable user code, that passes around strings as const, not compile.